utils/scarb-ui/src/verbosity.rs

```diff
@@ -14,6 +14,10 @@ pub enum Verbosity {
     ///
     /// String representation: `quiet`.
     Quiet,
+    /// Avoid printing warnings to standard output.
+    ///
+    /// String representation: `no-warnings`.
+    NoWarnings,
```

Comment:
This can be
```suggestion
    Error,
```
Indicating that it will print errors but not warnings

Reply:
That was my initial thought too, but we have some outputs that aren’t warnings or errors, like:
```rs
    ws.config()ui().print(Status::new("Packaging", &pkg_id.to_string()));
```
I figured we probably want to keep these messages and only mute warnings, so `NoWarnings` seemed more fitting.

Reply:
Yes, I would still use `ERROR` as it's more customary

Reply:
Are you sure it's more customary? Unlike `--quiet`, `--verbose` and `--no-warnings`, `--error` imo doesn't seem like a flag directly related to verbosity, and reading docs will be required to understand what it actually does.

Reply:
Hmm, I see what you mean, but personally I would stick with Error even if this does not fully reflect what we do, instead of using a negation 🤔 wdyt?

Reply:
I think it's useful for the flag to be directly related to what we do. 

> instead of using a negation 🤔 wdyt?

I'm not sure I see a problem in negation per se, we use it all the time already: `--no-build`, `--no-verify`, `--no-deps`, `--no-color` - the latter is directly related to output same as `--no-warnings`. 

For me, CURL comes to mind with:
- `--silent` -  hides progress / info output, i.e. hides all messages.
- `--show-error` - ensures errors are printed despite silence, while rest of the messages are silent.

I feel like if flag `--show-error` would actually keep both errors **and** regular messages, it would be misleading. Similarly here, I fee like `--error` flag would be misleading.

Thoughts?

Reply:
I don't have a problem with `--no-warnings` CLI flag - it serves the purpose well. However, I don't particularly like the `NoWarnings` level of `Verbosity` (i.e. `SCARB_UI_VERBOSITY` env value). I still fell like usually verbosity levels are modeled around nouns and not negations (e.g. https://docs.rs/clap-verbosity/latest/clap_verbosity/enum.Level.html).

Reply:
> I still fell like usually verbosity levels are modeled around nouns and not negations (e.g. https://docs.rs/clap-verbosity/latest/clap_verbosity/enum.Level.html).

Sure, but they mean something different in this context - if logging level is set to `Error`, it's expected **nothing** is printed except for errors.

If we're intending to keep logging level structure similar to the one you linked, we might consider handling `--no-warnings` separately altogether (with a separate env var as well).

We might want to add `ERROR` level in the future, and it would filter both everything that's below error level. 
Otherwise, having `SCARB_UI_VERBOSITY = ERROR` right now that is not behaving how you would **normally** expect it to seems misleading to me.

Reply:
> if logging level is set to Error, it's expected nothing is printed except for errors

I don't agree with this statement. If logging verbosity is set to `WARN` it doesn't mean that you only print warnings, but warning and errors. And for `INFO`, you would print three types of messages, etc. 
We categorize our messages into levels of verbosity and impose some arbitrary ordering between them. 
If "ERROR" sounds misleading to you, maybe `PRINT` or something like this? 
Otherwise, by extending your logic, the next level should be `NoWarningsNoErrorsNoPrints`, etc. 

> If we're intending to keep logging level structure similar to the one you linked, we might consider handling --no-warnings separately altogether (with a separate env var as well).

I thought about it as well, but IMHO it complicates this on both ends unnecessarily. 

>  Also, we might want to add level in the future that would filter everything that's below error level, and there's no other name than ERROR for that 😅

You can suggest different noun :D Apart from this very philosophical discussion the PR looks ready to me :D